src/commands/pickFuncProcess.ts

```diff
@@ -7,26 +7,25 @@ import * as unixPsTree from 'ps-tree';
 import * as vscode from 'vscode';
 import { IActionContext, UserCancelledError } from 'vscode-azureextensionui';
 import { extensionPrefix, funcHostStartCommand, isWindows } from '../constants';
+import { IPreDebugValidateResult, preDebugValidate } from '../debug/validatePreDebug';
 import { isFuncHostTask, stopFuncHost } from '../funcCoreTools/funcHostTask';
-import { validateFuncCoreToolsInstalled } from '../funcCoreTools/validateFuncCoreToolsInstalled';
 import { localize } from '../localize';
 import { getWindowsProcessTree, IProcessTreeNode, IWindowsProcessTree } from '../utils/windowsProcessTree';
-import { getDebugConfigs, isDebugConfigEqual } from '../vsCodeConfig/launch';
 import { getWorkspaceSetting } from '../vsCodeConfig/settings';
 
 export async function pickFuncProcess(this: IActionContext, debugConfig: vscode.DebugConfiguration): Promise<string | undefined> {
-    if (!await validateFuncCoreToolsInstalled()) {
+    const result: IPreDebugValidateResult = await preDebugValidate(debugConfig);
+    if (!result.shouldContinue) {
         throw new UserCancelledError();
```

Comment:
Since result.shouldContinue can be false without user input, should we throw an error that the user can action upon? (such as opening a workspace)

Reply:
How can it be false without user input?

Reply:
https://github.com/Microsoft/vscode-azurefunctions/blob/802c350b194543dfd35e3c02c0087e5bd06a224c/src/commands/createNewProject/verifyIsProject.ts#L28-L53

Doesn't this return `undefined` if there isn't a workspace opened?

Reply:
You can't get to pickProcess if you don't have a workspace open since it only happens when you debug